### PR TITLE
Webshell hunting query

### PR DESCRIPTION
**Pull request by vaniMSTIC**
Fixes #

## Proposed Changes

- Webshell Hunting Query

### CODE REVIEW

File: Hunting Queries/W3CIISLog/WebShellActivity.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  W3CIISLog 
+  | where TimeGenerated between ((start_time - lookback_time_from_start_time).. start_time)
+  | where not(ipv4_is_private(cIP)) and cIP != "127.0.0.1"
+  | summarize count() by cIP, csUserAgent
+  | project cIP, csUserAgent
+  );
+  let potential_webshell_activity = (W3CIISLog
+  | where TimeGenerated between (start_time .. end_time)
+  | extend csUriStem = tolower(csUriStem)
+  | where csUriStem matches regex \\.[a-zA-Z][a-zA-Z0-9]+$
```
Comment (petebryan) [inline Hunting Queries/W3CIISLog/WebShellActivity.yaml:35]: Your regex needs to be enclosed in quotes here

**Review by petebryan — CHANGES_REQUESTED** [review]

File: Hunting Queries/W3CIISLog/WebShellActivity.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  - InitialAccess
+relevantTechniques:
+  - T1100
+query: | 
+
+  let end_time = now();
+  let start_time = end_time - timespan(1d);
+  let lookback_time_from_start_time = timespan(3d);
+  let script_extensions = dynamic([".asp", ".aspx", ".armx", ".asax", ".ashz", ".asmx", ".axd", ".cshtml", ".php", ".phps", ".php3", ".php4", ".php5", ".php7", ".jsp", ".jspx", ".cfm", ".cfml", ".phtml"]);
+  let ignore_uristems = dynamic(["/ews/exchange.asmx"]);
```
[the remark this answers is not in the document]
Comment (vaniMSTIC) [reply Hunting Queries/W3CIISLog/WebShellActivity.yaml:24]: Done

File: Hunting Queries/W3CIISLog/WebShellActivity.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+id: e0c947c3-fe83-46ff-bbda-a43224a785fd
+name: Web Shell Activity
+description: |
+  'Web shells are scripts that, when uploaded to a web server, can be used to provide a backdoor to a compromised network.
+  Attackers can use this entry point to leave malicious implants, such as obtaining unauthorized access, escalating privilege, and further compromising the environment. 
+  
+  This query hunts for web shells by analysing the distribution of commonly-used web shell scripts againts regular scripts for those public client IPs which have not observed any W3CIIS activity in a fixed lookback period.
```
[the remark this answers is not in the document]
Comment (vaniMSTIC) [reply Hunting Queries/W3CIISLog/WebShellActivity.yaml]: Resolved

File: Hunting Queries/W3CIISLog/WebShellActivity.yaml (hunk 1 above)
Comment (vaniMSTIC) [reply Hunting Queries/W3CIISLog/WebShellActivity.yaml]: Resolved

**Review by petebryan — APPROVED** [review]

**Pull request merged**